### PR TITLE
[node] Accept readonly arguments

### PR DESCRIPTION
**Pull request by jablko**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

**Comment by andrewbranch** [comment]
This is usually a good idea but definitely want to let owners have a chance to review. In the meantime, it’s worth adding some tests for this one just to ensure the change persists through future updates.

**Comment by peterblazejewicz** [comment]
@jablko
I don't think going back and making change for every version is a good way here. IMO we should focus on LTS versions, to let people grasp the change impact. So at least v10, no chanes for v0, v4, etc.

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-calendar ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-calendar@latest/))
was missing the following properties:
1. Calendar
## react-select ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-select@latest/))
was missing the following properties:
1. defaultTheme
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c544e557a2e6da361e8efdf7999721f84efb925a
</p>

File: types/node/test/buffer.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ const result2 = Buffer.concat([utf8Buffer, base64Buffer], 9999999);
 // Class Method: Buffer.from(data)
 {
     // Array
-    const buf1: Buffer = Buffer.from([0x62, 0x75, 0x66, 0x66, 0x65, 0x72]);
+    const buf1: Buffer = Buffer.from([0x62, 0x75, 0x66, 0x66, 0x65, 0x72] as ReadonlyArray<number>);
```
Comment (ExE-Boss) [inline types/node/test/buffer.ts:44]: You can use `as const` to do this:
```suggestion
    const buf1: Buffer = Buffer.from([0x62, 0x75, 0x66, 0x66, 0x65, 0x72] as const);
```

Comment (jablko) [reply types/node/test/buffer.ts:44]: :+1: Absolutely, `as const` is the right thing to do here, `as ReadonlyArray<...>` is just so the tests pass on TypeScript 3.2.

Comment (jablko) [reply types/node/test/buffer.ts:44]: ... however where `typesVersions >= 3.4` I should do as you suggest, thanks! Done :heavy_check_mark:

**Comment by orta** [comment]
This PR seems solid to me as an immutable array is a subset of the array, so I don't really expect this to be breaking. 

We DT maintainers have shuffled it along 4 times without raising a blocking review, so I'm going to call this as reasonable and merge.

```ts
function funcWithReadonly(args: ReadonlyArray<string>) {}

funcWithReadonly([])

const b = []
b.push("")
funcWithReadonly(b)

const c: string[] = []
funcWithReadonly(c)
```

**Pull request merged**